bridge/mattermost/mattermost.go

```diff
@@ -715,19 +716,42 @@ func (m *Mattermost) wsActionPostSkip(rmsg *model.WebSocketEvent) bool {
 		return true
 	}
 
-	if data.UserId == m.GetMe().User {
-		if _, ok := extraProps["matterircd_"+m.GetMe().User].(bool); ok {
-			logger.Debugf("message is sent from matterirc, not relaying %#v", data.Message)
-			return true
-		}
+	if data.UserId != m.GetMe().User {
+		return false
+	}
+
+	if _, ok := extraProps["matterircd_"+m.GetMe().User].(bool); !ok {
+		return false
+	}
+
+	if data.Type == model.PostTypeLeaveChannel || data.Type == model.PostTypeJoinChannel {
+		logger.Debugf("our own join/leave message. not relaying %#v", data.Message)
+		return true
+	}
+
+	msgID := data.Id
+	msg := data.Message
+	channel := m.GetChannelName(data.ChannelId)
 
-		if data.Type == model.PostTypeLeaveChannel || data.Type == model.PostTypeJoinChannel {
-			logger.Debugf("our own join/leave message. not relaying %#v", data.Message)
-			return true
+	if strings.Contains(channel, "__") {
+		receiver := m.getDMUser(channel)
+		channel = receiver.Username
+	}
+
+	if data.RootId != "" {
+		msgID = data.RootId
+		if !m.v.GetBool("mattermost.hidereplies") {
+			newMsg, err := m.addParentMsg(data.RootId, data.Message, m.v.GetInt("mattermost.ShortenRepliesTo"), "@", m.v.GetBool("mattermost.unicode"))
+			if err == nil {
+				msg = newMsg
+			}
 		}
 	}
 
-	return false
+	m.msgLastSentCache.Add(msgID, fmt.Sprintf("%s: %s", channel, msg))
```

Comment:
wrt to line 732 to 751
I must be missing something but why do you want to add a message that's going to be skipped in to the cache?

Reply:
This is the only way we can get the message IDs for these messages we've sent. It goes through the server and back to us and that's when we add it to the cache.